src/modules/analytics/analytics.service.ts

```diff
@@ -212,28 +159,81 @@ export class AnalyticsService {
     if (events.length === 0) {
       return;
     }
-
+    this.data = [];
     for (const rawEvent of events) {
       try {
         let parsedEvent = undefined;
-        if (rawEvent.identifier === 'acceptOffer') {
-          parsedEvent = await this.acceptEventParser.handle(rawEvent);
-        } else {
-          parsedEvent = await this.buyEventHandler.handle(rawEvent, 'hash');
+        switch (rawEvent.identifier) {
```

Comment:
I would not use the event identifier here. Use the first topic from event which represents exactly the event name. Identifier is the endpoint name that triggered the event and you could have different events triggered from the same endpoint

Reply:
that's right, but I have mapped the enums based on the endpoint name.